README.MD

```diff
@@ -1,224 +1,277 @@
 
-```bash
-export NAMESPACE=manticore
-export BALANCER_NAME=$(kubectl get pods --namespace $NAMESPACE \
--l "app.kubernetes.io/name=manticoresearch,app.kubernetes.io/instance=manticore,name=manticore-manticoresearch-balancer" \
--o jsonpath="{.items[0].metadata.name}")
-```
+#### Configuration Options:
+- **Custom Behavior:** You can customize the balancer’s behavior using its HA strategy (`index_ha_strategy`).
+- **Enhanced Performance:** Scaling the number of balancer replicas improves the cluster’s ability to handle read requests.
 
-| SQL | HTTP | 
-|---|---|
-|`kubectl --namespace $NAMESPACE port-forward $BALANCER_NAME 6306:9306`|`kubectl --namespace $NAMESPACE port-forward $BALANCER_NAME 6308:9308`|
+For more details, refer to the [Load Balancing documentation](https://manual.manticoresearch.com/dev/Creating_a_cluster/Remote_nodes/Load_balancing#Load-balancing).
 
-### Search through the balancer:
+### How Does Auto OPTIMIZE Work?
 
-| SQL | HTTP | 
-|---|---|
-|`mysql -h0 -P6306 -e "SELECT * FROM idx WHERE match('dog')"`|`curl -X "POST" localhost:6308/search -d '{"index" : "idx", "query": { "match": { "_all": "dog" } } }'`|
+To enable automatic index compaction, update `values.yaml` and set `optimize.enabled = true`. The `optimize.interval` parameter controls how often disk chunk merging attempts occur.
 
-### Scale the cluster:
+The auto OPTIMIZE process follows this algorithm:
+1. Only one compaction session (OPTIMIZE) is allowed at a time. It checks if the previous OPTIMIZE session has finished or never started.
+2. If all conditions are met, it retrieves information about pods labeled as `component=worker + instance` and performs OPTIMIZE on each pod sequentially.
+3. Selects the index that benefits the most from compaction.
+4. Determines the number of CPU cores available to calculate the optimal chunk count to retain after compaction:
+   - Uses pod’s CPU limits if set.
+   - Defaults to host CPU cores if limits are not set.
+5. Executes OPTIMIZE with `cutoff=<cpuCores * values.optimize.coefficient>`.
 
-```
-kubectl scale statefulsets manticore-manticoresearch-worker -n manticore --replicas=5
-```
+### How does replication work?
+The replication script affects only worker pods. Each worker does the following every time it starts:
+![Replication workflow](replication.jpg)
 
-## Q&A:
-#### Where is the data actually stored?
-In volumes. One volume per worker. The PVC is by default 1 gigabyte (see variable `worker.volume.size`)
+### What's the Difference Between the Replication Modes?
 
-#### How is high availability achieved?
-* synchronous master-master replication
-* data remains in volume even after the worker is stopped/deleted
+The Manticore Search Helm chart supports two replication modes: **master-to-master** and **master-to-slave**. These modes determine how a worker pod behaves when it joins a cluster or reconnects after being disconnected.
```

Comment:
Add info about the 3rd mode - without using the distributed tables at all.

Reply:
No, this section describes replication between workers, connection to nodes after a restart, etc. It does not involve accessing the balancer. Essentially, it explains which master node a new node will join after being created or restarted.

So IMHO mentions about Balancer (distributed tables) here is unnessesarry